theano/gpuarray/subtensor.py

```diff
@@ -552,7 +552,7 @@ def perform(self, node, inputs, out_):
         # build the strides
         strides = [1]
         for i in range(p - 1, 0, -1):
-            stride = x.shape[i] * strides[-1]
```

Comment:
I'm not entirely convinced that the new code is better, but if the test passes that's good for me.